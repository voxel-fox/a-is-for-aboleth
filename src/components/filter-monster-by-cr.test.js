import {
  getCRValue,
  FilterMonsterByCR,
  toolTipFormat
} from "./filter-monster-by-cr";
import { act } from "react-dom/test-utils";

import { Range } from "rc-slider";

describe("getCRValue", () => {
  it("calculates CRs correctly", () => {
    expect(getCRValue(0)).toEqual(0);
    expect(getCRValue(1)).toEqual(0.125);
    expect(getCRValue(2)).toEqual(0.25);
    expect(getCRValue(3)).toEqual(0.5);
    expect(getCRValue(4)).toEqual(1);
    expect(getCRValue(33)).toEqual(30);
  });
});

describe("toolTipFormat", () => {
  it("formats tooltip values correctly", () => {
    expect(toolTipFormat(0)).toEqual("0");
    expect(toolTipFormat(1)).toEqual("1/8");
    expect(toolTipFormat(2)).toEqual("1/4");
    expect(toolTipFormat(3)).toEqual("1/2");
    expect(toolTipFormat(4)).toEqual("1");
    expect(toolTipFormat(33)).toEqual("30");
  });
});

describe("<FilterMonsterByCR />", () => {
  it("renders correctly", () => {
    const wrapper = mount(<FilterMonsterByCR />);
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onValueChange", () => {
    const onValueChange = sinon.spy();
    const wrapper = mount(<FilterMonsterByCR onValueChange={onValueChange} />);
    const onChange = wrapper.find(Range).prop("onChange");
    act(() => {
      onChange([1, 33]);
    });
    expect(
      onValueChange.calledOnceWith({
        min: 0.125,
        max: 30
      })
    ).toBe(true);
  });
});
