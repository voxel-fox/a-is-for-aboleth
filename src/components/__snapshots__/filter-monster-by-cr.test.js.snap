// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterMonsterByCR /> renders correctly 1`] = `
<FilterMonsterByCR>
  <div
    css={
      Object {
        "map": undefined,
        "name": "375sc5-wrapStyle",
        "next": undefined,
        "styles": "max-width:100%;width:21.25rem;overflow:visible;label:wrapStyle;",
      }
    }
  >
    <p
      css={
        Object {
          "map": undefined,
          "name": "1536rmu-toolTipStyle",
          "next": undefined,
          "styles": "text-align:center;margin:0 0 -0.3125rem 0;color:#999;label:toolTipStyle;",
        }
      }
    >
      <abbr
        title="Challenge Rating"
      >
        CR
      </abbr>
      :
      0
       -
       
      30
    </p>
    <ComponentEnhancer(Range)
      activeDotStyle={Object {}}
      allowCross={false}
      className=""
      count={1}
      css={
        Object {
          "map": undefined,
          "name": "oh60rc-sliderStyle",
          "next": undefined,
          "styles": "margin:0.625rem;max-width:calc(100% - 1.25rem);label:sliderStyle;",
        }
      }
      defaultValue={
        Array [
          0,
          33,
        ]
      }
      disabled={false}
      dotStyle={Object {}}
      dots={false}
      handle={[Function]}
      handleStyle={
        Array [
          Object {
            "backgroundColor": "#D8D8D8",
            "border": 0,
            "boxShadow": "0 2px 4px 1px rgba(0,0,0,0.50)",
            "height": 18,
            "marginLeft": -9,
            "marginTop": -6,
            "width": 18,
          },
        ]
      }
      included={true}
      marks={Object {}}
      max={33}
      min={0}
      onAfterChange={[Function]}
      onBeforeChange={[Function]}
      onChange={[Function]}
      prefixCls="rc-slider"
      pushable={false}
      railStyle={
        Object {
          "backgroundColor": "#666666",
          "height": 3,
        }
      }
      step={1}
      tabIndex={Array []}
      trackStyle={
        Array [
          Object {
            "backgroundColor": "#CACACA",
            "height": 3,
          },
        ]
      }
      vertical={false}
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#666666",
              "height": 3,
            }
          }
        />
        <Track
          className="rc-slider-track rc-slider-track-1"
          included={true}
          key="1"
          length={100}
          offset={0}
          style={
            Object {
              "backgroundColor": "#CACACA",
              "height": 3,
            }
          }
          vertical={false}
        >
          <div
            className="rc-slider-track rc-slider-track-1"
            style={
              Object {
                "backgroundColor": "#CACACA",
                "height": 3,
                "left": "0%",
                "width": "100%",
              }
            }
          />
        </Track>
        <Steps
          activeDotStyle={Object {}}
          dotStyle={Object {}}
          dots={false}
          included={true}
          lowerBound={0}
          marks={Object {}}
          max={33}
          min={0}
          prefixCls="rc-slider"
          step={1}
          upperBound={33}
          vertical={false}
        >
          <div
            className="rc-slider-step"
          />
        </Steps>
        <Handle
          className="rc-slider-handle rc-slider-handle-1"
          disabled={false}
          key="0"
          max={33}
          min={0}
          offset={0}
          prefixCls="rc-slider"
          style={
            Object {
              "backgroundColor": "#D8D8D8",
              "border": 0,
              "boxShadow": "0 2px 4px 1px rgba(0,0,0,0.50)",
              "height": 18,
              "marginLeft": -9,
              "marginTop": -6,
              "width": 18,
            }
          }
          tabIndex={0}
          value={0}
          vertical={false}
        >
          <div
            aria-disabled={false}
            aria-valuemax={33}
            aria-valuemin={0}
            aria-valuenow={0}
            className="rc-slider-handle rc-slider-handle-1"
            onBlur={[Function]}
            onKeyDown={[Function]}
            role="slider"
            style={
              Object {
                "backgroundColor": "#D8D8D8",
                "border": 0,
                "boxShadow": "0 2px 4px 1px rgba(0,0,0,0.50)",
                "height": 18,
                "left": "0%",
                "marginLeft": -9,
                "marginTop": -6,
                "width": 18,
              }
            }
            tabIndex={0}
          />
        </Handle>
        <Handle
          className="rc-slider-handle rc-slider-handle-2"
          disabled={false}
          key="1"
          max={33}
          min={0}
          offset={100}
          prefixCls="rc-slider"
          tabIndex={0}
          value={33}
          vertical={false}
        >
          <div
            aria-disabled={false}
            aria-valuemax={33}
            aria-valuemin={0}
            aria-valuenow={33}
            className="rc-slider-handle rc-slider-handle-2"
            onBlur={[Function]}
            onKeyDown={[Function]}
            role="slider"
            style={
              Object {
                "left": "100%",
              }
            }
            tabIndex={0}
          />
        </Handle>
        <Marks
          className="rc-slider-mark"
          included={true}
          lowerBound={0}
          marks={Object {}}
          max={33}
          min={0}
          onClickLabel={[Function]}
          upperBound={33}
          vertical={false}
        >
          <div
            className="rc-slider-mark"
          />
        </Marks>
      </div>
    </ComponentEnhancer(Range)>
  </div>
</FilterMonsterByCR>
`;
